pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -229,7 +231,15 @@ type ReconcileDatadogAgentDeployment struct {
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+//
+// Reconcile wraps internelReconcile to send metrics based on reconcile errors
 func (r *ReconcileDatadogAgentDeployment) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	resp, err := r.internelReconcile(request)
+	r.forwarders.ProcessError(request.NamespacedName, err)
```

Comment:
Do you need to check for `err != nil` before calling `ProcessError`?

Reply:
we're interested in processing all errors whatever their values, a `nil` error indicates a reconcile success

Reply:
I thought that success maps to `ProcessEvent`?

Reply:
`ProcessEvent` only converts k8s events into datadog events -> sends events
`ProcessError` sends a metric